torch/csrc/autograd/function.h

```diff
@@ -578,7 +578,7 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
 
   // Used by compiled autograd to call apply() with different saved tensors
   // Implementations should call saved.before() on all attrs, then apply(), then
-  // saved.after() on all attrs.
+  // saved.after() on all attrs in the same order.
```

Comment:
Random idea: When building with DEBUG, add extra metadata to tag every entry to ensure that there are no errors in the ordering.